diff --git a/include/target.mk b/include/target.mk
index 1b32bf905d..fd9b800787 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -50,7 +50,7 @@ DEFAULT_PACKAGES.nas:=\
 # @brief Default packages for @DEVICE_TYPE router.
 ##
 DEFAULT_PACKAGES.router:=\
-	dnsmasq \
+	dnsmasq-full luci luci-app-ttyd \
 	firewall4 \
 	nftables \
 	kmod-nft-offload \
diff --git a/target/linux/generic/config-6.6 b/target/linux/generic/config-6.6
index 9eec780b53..bf21156cc3 100644
--- a/target/linux/generic/config-6.6
+++ b/target/linux/generic/config-6.6
@@ -3126,7 +3126,7 @@ CONFIG_LINEAR_RANGES=y
 # CONFIG_LIQUIDIO is not set
 # CONFIG_LIQUIDIO_VF is not set
 # CONFIG_LIRC is not set
-CONFIG_LIST_HARDENED=y
+# CONFIG_LIST_HARDENED is not set
 # CONFIG_LITEX_LITEETH is not set
 # CONFIG_LITEX_SOC_CONTROLLER is not set
 # CONFIG_LIVEPATCH is not set
diff --git a/target/linux/ramips/mt7621/target.mk b/target/linux/ramips/mt7621/target.mk
index 2b9a1312af..e428774642 100644
--- a/target/linux/ramips/mt7621/target.mk
+++ b/target/linux/ramips/mt7621/target.mk
@@ -10,7 +10,7 @@ KERNELNAME:=vmlinux vmlinuz
 # make Kernel/CopyImage use $LINUX_DIR/vmlinuz
 IMAGES_DIR:=../../..
 
-DEFAULT_PACKAGES += wpad-basic-mbedtls uboot-envtools kmod-crypto-hw-eip93
+DEFAULT_PACKAGES += -wpad-basic-mbedtls wpad-openssl uboot-envtools kmod-crypto-hw-eip93
 
 define Target/Description
 	Build firmware images for Ralink MT7621 based boards.
